crates/redis/src/lib.rs

```diff
@@ -107,15 +110,29 @@ impl TriggerExecutor for RedisTrigger {
 impl RedisTrigger {
     // Handle the message.
     async fn handle(&self, msg: redis::Msg) -> Result<()> {
-        let channel = msg.get_channel_name();
+        let channel = msg.get_channel_name().to_owned();
         tracing::info!("Received message on channel {:?}", channel);
 
-        if let Some(component_id) = self.channel_components.get(channel) {
-            tracing::trace!("Executing Redis component {component_id:?}");
-            let executor = SpinRedisExecutor;
-            executor
-                .execute(&self.engine, component_id, channel, msg.get_payload_bytes())
-                .await?
+        if let Some(component_ids) = self.channel_components.get(&channel) {
+            let futures: Vec<_> = component_ids
+                .iter()
+                .map(|id| {
+                    tracing::trace!("Executing Redis component {id:?}");
+                    SpinRedisExecutor.execute(&self.engine, id, &channel, msg.get_payload_bytes())
+                })
+                .collect();
+            let results: Vec<_> = join_all(futures).await.into_iter().collect();
+            let mut errors = Vec::new();
+            for result in results {
+                if result.is_err() {
+                    if let Some(err) = result.err() {
+                        errors.push(err)
+                    }
+                }
+            }
```

Comment:
This is a bit more terse and IMO readable:

```suggestion
            let errors = results.filter_map(|r| r.err()).collect::<Vec<_>>():
           
```

Or you could just return the first error:

```suggestion
           results.collect::<Result<Vec<_>>()?:
```

Reply:
Which would be the preferred approach here?

Reply:
The first option is probably a little better but it doesn't matter much because errors are only checked in tests.

Speaking of which, we should probably log errors here :sweat_smile:

Reply:
Okay I went with the first option.